doc/source/contribution_and_api/unit_testing.rst

```diff
@@ -0,0 +1,42 @@
+.. _ref_unit_testing_contributing:
+
+Unit Testing
+===============
+
+Unit testing is highly important. It enables to check that the
+changes on the code are efficient and don't have an impact on
+other parts of the code. 
+
+Unit tests are in the ``tests`` directory inside this repository.
+
+Test Validation GitHub workflow
+-------------------------------
+
+`Codecov <https://github.com/codecov>`_ generates a coverage report of
+the committed code. It details how merging the Pull Request will impact the
+coverage. It is one of the checks that must be approved to
+merge the code.
```

Comment:
I think this does not belong to this section.

I would put at the end of the first block... try to link it with the previous text by telling why coverage is needed.